src/ui/components/Dashboard/Dashboard.jsx

```diff
@@ -1,61 +1,151 @@
+} from '@material-ui/core';
+import MuiAlert from '@material-ui/lab/Alert';
 import useStyles from './styles';
-import Sidebar from './Sidebar';
-import NavBar from './NavBar';
-import Collections from '../Collections';
+import usePlanDefinitions from '../../hooks/usePlanDefinitions';
+import axios from 'axios';
+
 function Dashboard() {
   const classes = useStyles();
-  const [contentKey, setContentKey] = useState('');
-  // can do useMemo to update labels like notifications
-  const tabs = [
-    { key: 'overview', label: 'Overview', component: null },
-    { key: 'dashboard', label: 'Dashboard', component: <div>Dashboard content</div> },
-    { key: 'notifications', label: 'Notifications', component: <div>Notifications</div> },
-    { key: 'collections', label: 'Collections', component: null },
-    { key: 'servers', label: 'Servers', component: <Collections selectedCollection="servers" /> },
-    {
-      key: 'endpoints',
-      label: 'Endpoints',
-      component: <Collections selectedCollection="endpoints" />
-    },
-    {
-      key: 'subscriptions',
-      label: 'Subscriptions',
-      component: <Collections selectedCollection="subscriptions" />
-    },
-    {
-      key: 'plandefinitions',
-      label: 'Plan Definitions',
-      component: <Collections selectedCollection="plandefinitions" />
+  const importFileRef = useRef(null);
+  const [fileName, setFileName] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [alertMessage, setAlertMessage] = useState('');
+  const [alertSeverity, setAlertSeverity] = useState('');
+  const [planDefFullUrl, setPlanDefFullUrl] = useState('');
+  const { data: planDefinitions } = usePlanDefinitions();
+
+  const handleChangePlanDefFullUrl = useCallback(
+    event => {
+      setPlanDefFullUrl(event.target.value);
     },
-    { key: 'logs', label: 'Logs', component: <Collections selectedCollection="logs" /> },
-    {
-      key: 'completedreports',
-      label: 'Completed Reports',
-      component: <Collections selectedCollection="completedreports" />
+    [setPlanDefFullUrl]
+  );
+
+  const handleChooseFile = useCallback(() => {
+    if (importFileRef?.current?.files?.[0]) setFileName(importFileRef.current.files[0].name);
+  }, [importFileRef, setFileName]);
+
+  const handleTrigger = useCallback(
+    event => {
+      event.preventDefault();
+      setIsLoading(true);
+      const files = importFileRef.current.files;
+      const reader = new global.FileReader();
+      reader.onload = e => {
+        const rawContent = e.target.result;
+        const resource = JSON.parse(rawContent);
+        axios
+          .post('/index/trigger', { planDefFullUrl, resource }, { withCredentials: true })
+          .then(() => {
+            setIsLoading(false);
+            setFileName('');
+            setPlanDefFullUrl('');
+            setAlert('Manual Report Triggered Successfully!', 'success');
+          })
+          .catch(err => {
+            console.error(err);
```

Comment:
I think we want to `setIsLoading(false)` here as well, so if there's an error it unlocks the button and the user can try again.